src/components/header/EmptyThreadIndicator.js

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React, { PureComponent } from 'react';
+import { withSize } from '../shared/WithSize';
+import Tooltip from '../shared/Tooltip';
+
+import type { Thread } from '../../types/profile';
+import type { Milliseconds, StartEndRange } from '../../types/units';
+
+import './EmptyThreadIndicator.css';
+
+type Props = {
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +width: number,
+  +thread: Thread,
+  +unfilteredSamplesRange: StartEndRange | null,
+};
+
+type State = {
+  hoveredItem: *,
+  mouseX: number,
+  mouseY: number,
+};
+
+class EmptyThreadIndicator extends PureComponent<Props, State> {
+  _canvas: HTMLCanvasElement | null;
+  _requestedAnimationFrame: boolean | null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      hoveredItem: null,
+      mouseX: 0,
+      mouseY: 0,
+    };
+  }
+
+  _getIndicatorPositions(): {
+    startup?: Object,
+    shutdown?: Object,
+    emptyBufferStart?: Object,
+    emptyBufferEnd?: Object,
+  } {
```

Comment:
Here, I think you can simply use `*` for the property types as Flow will very easily find out from the method's body.

Reply:
Or maybe you'd like to better type this in the future and this is a placeholder :)

Reply:
I guess this is really what it is:
```
type SyntheticCssDeclarations = {
  [string]: string | number,
};
```

Reply:
Don't you have fixed properties ?